packages/core/src/components/AvatarGroup/AvatarGroup.tsx

```diff
@@ -0,0 +1,155 @@
+
+import { ExtractNames } from "@core/utils/classes";
+
+import { staticClasses, useClasses } from "./AvatarGroup.styles";
+
+import { HvAvatar } from "../Avatar/Avatar";
+
+export { staticClasses as avatarGroupClasses };
+
+export type HvAvatarGroupClasses = ExtractNames<typeof useClasses>;
+
+export interface HvAvatarGroupProps extends HvBaseProps {
+  /** Inline styles to be applied to the root element. */
+  style?: CSSProperties;
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvAvatarGroupClasses;
+  /** The avatar size. */
+  size?: HvSize;
+  /** The spacing between avatars. */
+  spacing?: "compact" | "loose";
+  /** The direction of the group. */
+  direction?: "row" | "column";
+  /** Whether the avatars display behind the previous avatar or on top. */
+  toBack?: boolean;
+  /**
+   * The maximum number of visible avatars. If there are more avatars then the value of this property, an added avatar will
+   * be added to the end of the list, indicating the number of hidden avatars.
+   */
+  maxVisible?: number;
+  /**
+   * What to show as an overflow representation.
+   * If `undefined` a default `HvAvatar` will be displayed along with a HvTooltip with the count of overflowing items.
+   * */
+  overflowComponent?: (overflowCount: number) => React.ReactNode;
+}
+
+const getSpacingValue = (
+  spacing: HvAvatarGroupProps["spacing"],
+  size: HvAvatarGroupProps["size"]
+) => {
+  switch (size) {
+    case "xs":
+      return spacing === "compact" ? 24 : 16;
+    case "sm":
+      return spacing === "compact" ? 30 : 18;
+    case "md":
+      return spacing === "compact" ? 36 : 20;
+    case "lg":
+      return spacing === "compact" ? 44 : 24;
+    case "xl":
+      return spacing === "compact" ? 72 : 34;
+    default:
+      return spacing === "compact" ? 30 : 18;
+  }
+};
+
+/**
+ * The AvatarGroup component is used to group multiple avatars.
+ */
+export const HvAvatarGroup = forwardRef<any, HvAvatarGroupProps>(
+  (props, ref) => {
+    const {
+      className,
+      style,
+      classes: classesProp,
+      children,
+      size = "sm",
+      spacing = "loose",
+      direction = "row",
+      toBack = true,
+      maxVisible = 3,
+      overflowComponent,
+      ...others
+    } = useDefaultProps("HvAvatarGroup", props);
+    const { classes, cx } = useClasses(classesProp);
+
+    const spacingValue = getSpacingValue(spacing, size);
+
+    const totalChildren = Children.count(children);
+    const zIndexMultiplier = toBack ? -1 : 1;
+    const willOverflow = totalChildren > maxVisible;
+
+    return (
+      <div
+        className={cx(
+          classes.root,
+          css({ flexDirection: direction }),
+          className
+        )}
+        ref={ref}
+        {...others}
+      >
+        {Children.map(children, (child: any, index: number) => {
+          if (index < maxVisible) {
+            return cloneElement(child, {
+              style: {
```

Comment:
Do you think it’s possible to leverage a bit more `classes` in the code to enable users to more easily override the styles if necessary? Also, I believe using `style` for our styles instead of `className` makes it more difficult to override the styles externally.

Reply:
These styles are dynamic and need to be calculated on the fly. I needed to use the `style` prop to ensure the `zIndex` was preserved and the avatars were stacked properly.

Reply:
Okay, makes sense!